File: Factory Method/Ruby/LoggerFactory.rb
require_relative "InfoLogger"
require_relative "WarningLogger"
require_relative "ErrorLogger"
require_relative "LoggerType"

class LoggerFactory
    def create(type)
        case type
            when LoggerType::InfoLoggerType
                return InfoLogger.new
            when LoggerType::WarningLoggerType
                return WarningLogger.new
            when LoggerType::ErrorLoggerType
                return ErrorLogger.new
            else
                return nil
        end
    end
end

File: Factory Method/Javascript/LoggerFactory.js
import ErrorLogger from "./ErrorLogger.js";
import InfoLogger from "./InfoLogger.js";
import WarningLogger from "./WarningLogger.js";
import LoggerType from "./LoggerType.js";

class LoggerFactory {
  create(type) {
    switch (type) {
      case LoggerType.InfoLogger:
        return new InfoLogger();

      case LoggerType.WarningLogger:
        return new WarningLogger();

      case LoggerType.ErrorLogger:
        return new ErrorLogger();
    
      default:
        return null;
        break;
    }
  }
}

export default LoggerFactory;